File: server/src/main/java/au/com/codeka/warworlds/server/util/Pair.java
package au.com.codeka.warworlds.server.util;

import com.google.common.base.Objects;

/**
 * Helper class that represents a pair of values.
 */
public class Pair<E, F> {
  public E one;
  public F two;

  public Pair() {
  }

  public Pair(E one, F two) {
    this.one = one;
    this.two = two;
  }

  @Override
  public int hashCode() {
    return Objects.hashCode(one, two);
  }

  @Override
  public boolean equals(Object obj) {
    if (obj == null || !(obj instanceof Pair)) {
      return false;
    }

    Pair<E, F> other = (Pair) obj;
    return Objects.equal(other.one, one) && Objects.equal(other.two, two);
  }
}

File: server/src/main/kotlin/au/com/codeka/warworlds/server/util/Pair.kt
package au.com.codeka.warworlds.server.util

import com.google.common.base.Objects

/** Helper class that represents a pair of values. */
class Pair<E, F>(var one: E, var two: F) {
  override fun hashCode(): Int {
    return Objects.hashCode(one, two)
  }

  override fun equals(other: Any?): Boolean {
    if (other == null || other !is Pair<*, *>) {
      return false
    }
    return Objects.equal(other.one, one) && Objects.equal(other.two, two)
  }
}
